src/data/int/parity.lean

```diff
@@ -228,4 +229,67 @@ by simp [*, (dec_trivial : ¬ 2 = 0)] with parity_simps
 example : ¬ even (25394535 : ℤ) :=
 by simp
 
+section neg_one_pow
+
+/-- The map `ℤ → ℤ` sending $n$ to $(-1) ^ n$. -/
+def neg_one_pow (n : ℤ) : ℤ := @has_pow.pow (units ℤ) ℤ _ (-1) n
```

Comment:
Is it beneficial to have this in ℤ and not `units ℤ`?

Reply:
It allows lemmas about `zsmul` to be applied.

Reply:
Which lemmas about zsmul do you want? The downside about your approach is that it does not allow lemmas about `units` smul to be applied.

Reply:
Mainly `category_theory.preadditive.zsmul_comp` and its friend.
I don't think we will need those units smul lemmas though, we will hardly ever talk about the multiplicative inverse of this, and `units ℤ` is quite a boring group itself.

Reply:
I think a lot of the lemmas here generalize to `units R` for any ring R though, with practically identical proofs. I'd rather we just proved those missing lemmas than introduce a new API.

Reply:
I've started adding those lemmas in #11447